### PR TITLE
Minor: Improves error checks in run_route_scripts/batch.sh

### PR DESCRIPTION
**Pull request by PureW**
# Issue

A minor tweak of the `run_route_scripts/batch.sh` script without changing it's behaviour. Simply adding a few sanity checks to have bash catch errors (a bit like enabling `-Werror` in gcc).

## Tasklist

 - [x] Add tests (ran script locally)
 - [ ] Review - you must request approval to merge any PR to master
 - [ ] Add #fixes with the issue number that this PR addresses
 - [x] Generally use squash merge to rebase and clean comments before merging
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] Update relevant [documentation](docs/README.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by PureW** [comment]
May want to check with someone running bsd-tooling (e.g. osx) if the long-opts work, bsd tends to prefer the short-opts. cc @danpaz

**Comment by dgearhart** [comment]
@PureW I am assuming that the `run.sh` script works the same?

File: run_route_scripts/batch.sh [hunk 1]
```diff
@@ -18,37 +19,29 @@ function usage() {
...
+readonly DEFAULT_OUTDIR=$(date +%Y%m%d_%H%M%S)_$(basename "${INPUT%.*}")
+readonly RESULTS_OUTDIR="results/${4:-$DEFAULT_OUTDIR}"
+mkdir --parent "${RESULTS_OUTDIR}"
 
 #turn the nice input into something parallel can parse for args
 TMP="$(mktemp)"
-cp -rp "${INPUT}" "${TMP}"
-for arg in $(valhalla_run_route --help | grep -o '\-[a-z\-]\+' | sort | uniq); do
+cp --recursive --preserve=mode,ownership,timestamps "${INPUT}" "${TMP}"
+for arg in $(build/valhalla_run_route --help | grep -o '\-[a-z\-]\+' | sort | uniq); do
```
Comment (kevinkreiser) [inline run_route_scripts/batch.sh]: so previously this assumed the binary was on your `PATH` now it requires you to run this from a certain location? im a little confused by this part tbh. isnt this script nested in a dir typically one down from the root. meaning it wouldnt see `build` but rather `../build`?

Comment (PureW) [reply run_route_scripts/batch.sh]: Yea, I was just digging into @dgearhart's comment about `run.sh` and realised both of these scripts rely on the binaries being installed on system path.

Let me restore that behaviour. I tweaked it in order to more easily try different versions without having to install the binaries to the system but I see now that's a larger change than expected.

**Comment by PureW** [comment]
> I am assuming that the run.sh script works the same?

@dgearhart I just ran `run.sh` with the updated code I just pushed.

**Comment by dgearhart** [comment]
> @dgearhart I just ran `run.sh` with the updated code I just pushed.

Thank you @PureW

**Pull request merged**